spec.bs

```diff
@@ -702,6 +707,13 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
     a {{TypeError}}.
   1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
     |signals|.
+1. If |config|["{{AuctionAdConfig/perBuyerCurrencies}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerCurrencies}}"]:
+  1. If the result of result of [=checking whether a string is a valid currency tag=] on |value| is false, [=exception/throw=] a {{TypeError}}.
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyers currency=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be the result of [=parsing an origin=] with |key|. If it fails, [=exception/throw=] a {{TypeError}}.

```

Comment:
Parsing an origin returns either an origin or failure, so we probably want to say something like "... If |buyer| is failure, ..."

Reply:
Done